core/src/main/java/google/registry/tools/UniformRapidSuspensionCommand.java

```diff
@@ -138,6 +148,21 @@ protected void initMutatingEppToolCommand() {
     } else {
       statusesToApply = URS_LOCKS;
     }
+
+    if (renewOneYear) {
+      setSoyTemplate(DomainRenewSoyInfo.getInstance(), DomainRenewSoyInfo.RENEWDOMAIN);
+      DomainBase domainBase = domain.get();
+      addSoyRecord(
```

Comment:
You need to add support for specifying `requestedByRegistrar` and `reason` in the `MetadataExtension` to the `DomainRenewFlow`, and then call said behavior in your soy template (using the same values that are passed to the update flow).  This can be part of a separate PR.